### PR TITLE
Changed the models, started Auth

### PR DESCRIPTION
**Pull request by Harshvartak**

### CODE REVIEW

File: placementApp/models.py [hunk 1]
```diff
@@ -1,3 +1,111 @@
...
+	    regex=r"^\+?6?\d{10,12}$", message="SAP ID must be valid"
+	)
+	sap_ID = models.CharField(
+	    validators=[sap_regex],
+	    max_length=12,
+	    blank=False,
+	    null=False,
+	    default=None,
+	    unique=True,
+	)
```
Comment (SakshiUppoor) [inline placementApp/models.py]: SAP ID field should be included in the User model rather than the Student model

Comment (kanishkshah) [reply placementApp/models.py]: Actually they should be in the Student model , User model should only hold common attributes. The TPO / Co-ordinators need not have a Sap ID field

Comment (kanishkshah) [reply placementApp/models.py]: Actually they should be in the Student model , User model should only hold common attributes. The TPO / Co-ordinators need not have a Sap ID field

File: placementApp/models.py [hunk 2]
```diff
@@ -1,3 +1,111 @@
...
+	username = models.CharField(max_length=30, unique=True)
+	date_joined = models.DateTimeField(verbose_name="date joined", auto_now_add=True)
+	last_login = models.DateTimeField(verbose_name="last login", auto_now=True)
+	is_admin = models.BooleanField(default=False)
+	is_active = models.BooleanField(default=True)
+	is_staff = models.BooleanField(default=False)
+	is_superuser = models.BooleanField(default=False)
+	ROLE_CHOICES = [("STUDENT", "Student"), ("CO", "Co-ordinator"), ("TPO", "TPO")]
+	role = models.CharField(max_length=7, blank=False, choices=ROLE_CHOICES)
+
```
Comment (parag1999) [inline placementApp/models.py]: There is no need to keep is_admin, is_staff. is_superuser since we already have role field

**Pull request merged**